### ISSUE TITLE
Changes to CMake tool user/workspace settings are not recognised

### ISSUE THREAD

**Issue by mojofunk** [vector-of-bool/vscode-cmake-tools#566]
### Brief Issue Summary

I installed the CMake tools extension for the first time yesterday but I had some trouble adding some user settings.

I just wanted to specify a toolchain file (vcpkg) and set a couple of options in the User Settings.

What I found very confusing and not at all apparent is what is required to pick up changes in that file. I finally discovered that changing the "kit" would pick up the changes.

I spent far more time figuring this out than I would have liked and found it a bit frustrating.

So at the very least I think some change needs to occur so that changes in the user/workspace files that affect the CMake configuration are picked up and a reconfigure occurs with the new parameters.

Another thing that would have really helped would be to print out some more information, like for instance the command being called to configure CMake with the parameters.

It would have also been helpful for me if there was a specific section in the documentation relating to setting configure options and perhaps also setting a toolchain file.

Otherwise CMake tools seems pretty nice so far, so thanks for all the work on it.

### Expected:

Changes made to user/workspace settings are applied the next time CMake is run.

### Apparent Behavior:

Changes made to user/workspace settings are not applied until the "kit" is changed. There are possibly other ways to trigger a reconfigure with the new settings but I haven't found one yet.

### Platform and Versions

VSCode Version: 1.28.1 (user setup) and CMake Tools version 1.1.2 on Windows 10.

**Comment by vector-of-bool** [comment]
I think I might understand the issue, but please correct me if I am wrong:

1. You modified the CMake configuration settings in `settings.json`, but _did not_ make any changes to any CMake files
2. You re-ran the build, but CMake did not re-configure. (vcpkg was not being used.)
3. You changed the Kit, and found that CMake configured and vcpkg was now available.

If my understanding is correct, I think I know the source of this bug and a probable fix. Do I understand your problem correctly?

Also, you should be able to see the configure command in the output if you turn up the CMake Tools logging level. I may be misremembering that, though.

**Comment by mojofunk** [comment]
Yes, I think you have summarized the issue. I did not make any changes to my CMake files. The configure process could not complete successfully (to locate/use Qt5) without specifying/setting a vcpkg toolchain file. 

I ended up getting it to work successfully for my situation by adding this to the User Settings:

    "cmake.configureSettings" : {
        "CMAKE_TOOLCHAIN_FILE" : "C:/Users/mojof/Data/devel/vcpkg/scripts/buildsystems/vcpkg.cmake",
        "BUILD_TESTS" : "ON",
        "BUILD_SINGLE_TESTS" : "ON"
    }

But it was not picking up and using the settings until I started trying different things to try and trigger a full reconfigure. It took a while to figure out that changing the "kit" would cause the new settings to be applied, and I wasn't even sure if what I had added was valid etc, which is why some more verbose output may have been useful. 

How can I turn up the CMake Tools logging level? Is that documented somewhere?

Thanks for you help.
